lib/player.js

```diff
@@ -3798,7 +3798,19 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
   addTextTrack(uri, language, kind, mime, codec, label) {
     // TODO: Add an actual error for this.
     if (this.loadMode_ == shaka.Player.LoadMode.SRC_EQUALS) {
-      shaka.log.error('Cannot add text when loaded with src=');
+      const trackElement = document.createElement('track');
+      trackElement.setAttribute('src', uri);
+      trackElement.setAttribute('label', label || '');
+      trackElement.setAttribute('kind', kind);
+      trackElement.setAttribute('srclang', language);
+      this.video_.appendChild(trackElement);
```

Comment:
I would prefer it if this extra track object could be cleaned up on unload.

Reply:
Done!